python/sglang/srt/tracing/trace.py

```diff
@@ -287,451 +286,470 @@ def trace_set_thread_info(
 
-def trace_req_start(
-    rid: str,
-    bootstrap_room: Optional[int] = None,
-    ts: Optional[int] = None,
-    role: Optional[str] = "null",
-    external_trace_header: Optional[Dict[str, str]] = None,
-):
-    if not tracing_enabled:
-        return
-
-    rid = str(rid)
-
-    ts = ts or __get_cur_time_ns()
+        self.start_time_ns = get_cur_time_ns()
+        self.is_copy = True
 
-    pid = threading.get_native_id()
-    if pid not in threads_info:
-        return
-
-    # create req context and root span
-    bootstrap_room = 0 if bootstrap_room is None else bootstrap_room
-    reqs_context[rid] = SglangTraceReqContext(
-        rid=rid,
-        start_time_ns=ts,
-        threads_context={},
-        bootstrap_room=bootstrap_room,
-        is_copy=False,
-    )
-
-    # create bootstrap room span
-    tracer = threads_info[pid].tracer
-    if str(bootstrap_room) not in remote_trace_contexts:
-        attrs = {"bootstrap_room": str(hex(bootstrap_room))}
-        external_trace_context = _trace_context_propagator.extract(
-            external_trace_header
-        )
-        bootstrap_room_span = tracer.start_span(
-            name=f"Bootstrap Room {hex(bootstrap_room)}",
+        trace_context = SGLangTracePropagateContext.instance_from_dict(trace_context)
+        if not trace_context:
+            self.tracing_enable = False
+        else:
+            self.root_span_context = trace_context.root_span_context
+
+        self.thread_context = self.__create_thread_context(self.start_time_ns)
+        if self.tracing_enable:
+            self.thread_context.last_span_context = trace_context.prev_span_context
+
+    def trace_req_start(
+        self,
+        ts: Optional[int] = None,
+        external_trace_header: Optional[Dict[str, str]] = None,
+    ):
+        if not self.tracing_enable:
+            return
+
+        ts = ts or get_cur_time_ns()
+
+        # create req context and root span
+        bootstrap_room = 0 if self.bootstrap_room is None else self.bootstrap_room
+        self.start_time_ns = ts
+
+        # create bootstrap room span
+        tracer = threads_info[self.pid].tracer
+        if str(bootstrap_room) not in remote_trace_contexts:
+            attrs = {"bootstrap_room": str(hex(bootstrap_room))}
+            external_trace_context = _trace_context_propagator.extract(
+                external_trace_header
+            )
+            bootstrap_room_span = tracer.start_span(
+                name=f"Bootstrap Room {hex(bootstrap_room)}",
+                start_time=ts,
+                attributes=attrs,
+                context=external_trace_context,
+            )
+            self.bootstrap_room_span = bootstrap_room_span
+            self.bootstrap_room_span_context = trace.set_span_in_context(
+                bootstrap_room_span
+            )
+        else:
+            self.bootstrap_room_span_context = remote_trace_contexts[
+                str(bootstrap_room)
+            ].root_span_context
```

Comment:
`bootstrap_room` is always 0 for mix(non-PD) deployment? so `remote_trace_contexts` would save only one key like:

```
["0": bootstrap_room_span_context]
```

Will this cause a chaos for mixed deployment? I did some test and found that I can't correctly propagate upstream trace_id to SGLang for mix deployment.